### PR TITLE
remove excessive static_casts in SWC reader

### PR DESCRIPTION
**Pull request by mgeplf**
* don't allow negative IDs, except for the special parent == -1, for the first sample

### CODE REVIEW

File: src/readers/morphologySWC.cpp [hunk 1]
```diff
@@ -14,30 +15,49 @@
 #include <morphio/properties.h>
 
 namespace {
+// It's not clear if -1 is the only way of identifying the root section.
+const int SWC_UNDEFINED_PARENT = -1;
+const unsigned int SWC_ROOT = 0xFFFFFFFD;
```
Comment (eleftherioszisis) [inline src/readers/morphologySWC.cpp:20]: Does this need to be a different "unknown" number? Could we also use here `0xFFFFFFFE` or is it required they are differentiated?

Comment (mgeplf) [reply src/readers/morphologySWC.cpp:20]: doesn't need to be different, but it helps w/ debugging, so I like to use different numbers.

File: src/readers/morphologySWC.cpp [hunk 2]
```diff
@@ -14,30 +15,49 @@
...
                           format,
-                          &sample.id,
+                          &id,
                           &int_type,
                           &sample.point[0],
                           &sample.point[1],
                           &sample.point[2],
                           &radius,
-                          &sample.parentId) == 7;
+                          &parentId) == 7;
```
Comment (eleftherioszisis) [inline src/readers/morphologySWC.cpp:47]: Maybe turn 7 into a constant?

Comment (mgeplf) [reply src/readers/morphologySWC.cpp:47]: this will be going away once I get rid of `scanf`

**Review by eleftherioszisis — APPROVED** [review]

**Pull request merged**